app/src/main/java/com/kennedy/peter/bloquery/firebase/User.java

```diff
@@ -0,0 +1,15 @@
+package com.kennedy.peter.bloquery.firebase;
+
+public class User {
+    public String UID, userName, email;
+    public void updateUserDetails(String UID, String userName, String email) {
+        this.UID = (UID == null) ? this.UID : UID;
+        this.userName = (userName == null) ? this.userName : userName;
+        this.email = (email == null) ? this.email : email;
+    }
+    public void nullifyUser() {
```

Comment:
We can discuss it today, but the point of letting the user pass null values
into updateUserDetails was that i didn't want to have to keep on grabbing
details of the user whenever they updated their info. The only thing
they'll pass null values for is when they change their username they dont
have to look up their email and UID to also pass in those values they can
just toss in null if they dont know the other details. Similarly, when i
add in the rest of the data like their Picture and Description if they only
edit one thing i thought it'd be nice to make it so they can just null
everything that they dont know. On hind sight its pretty unnecessary since
im the only one seeing the code but this is just what made immediate sense
to me at the time.

ANd the use case of nullifyUser() is when they log out, i need to wipe the
user object clean for when they log into a different account.

On Fri, Feb 12, 2016 at 11:56 AM, Andrei Catinean <notifications@github.com>
wrote:

> In app/src/main/java/com/kennedy/peter/bloquery/firebase/User.java
> <https://github.com/pykennedy/BloQuery/pull/3#discussion_r52790521>:
>
> > @@ -0,0 +1,15 @@
> > +package com.kennedy.peter.bloquery.firebase;
> > +
> > +public class User {
> > +    public String UID, userName, email;
> > +    public void updateUserDetails(String UID, String userName, String email) {
> > +        this.UID = (UID == null) ? this.UID : UID;
> > +        this.userName = (userName == null) ? this.userName : userName;
> > +        this.email = (email == null) ? this.email : email;
> > +    }
> > +    public void nullifyUser() {
>
> What's the usecase of this method?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/pykennedy/BloQuery/pull/3/files#r52790521>.
>